Core/Src/gnwmanager.c

```diff
@@ -129,6 +149,37 @@ struct gnwmanager_comm {  // Values are read or written by the debugger
 
 static struct gnwmanager_comm comm __attribute__((section (".gnwmanager_comm")));
 
+static FATFS FatFs;  // Fatfs handle
+static FIL file; // File handle
+
+void sdcard_hw_detect() {
```

Comment:
maybe have this be:

```
gnwmanager_sdcard_hw_t sdcard_hw_detect()
```

and have the caller set the global `sdcard_hw` instead of doing it inside this function.

Reply:
The mount process used to detect if a SD Card is present and to detect which hardware is used is using the sdcard_hw global variable value, so it is changed during detection process before doing the mount. Making things cleaner will require some work ... Maybe it could be improved later ? 😬